### PR TITLE
Fix retrieval example issue

### PR DESCRIPTION
**Pull request by lvliang-intel**
## Type of Change

bug fix
API not changed

## Description

Fix retrieval example issue

## Expected Behavior & Potential Risk

The retrieval example can be run successfully.

## How has this PR been tested?

Pre-CI and UT.

## Dependency Change?

No.

### CODE REVIEW

**Review by hshen14 — APPROVED** [review]

**Comment by XuhuiRen** [comment]
the name of "run_retrieval_on_cpu.sh" may lead to unexpected confusion when the device has both GPU and CPU. Need to rename or specify the device in the code.

File: intel_extension_for_transformers/neural_chat/examples/retrieval/run_retrieval_on_cpu.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# KMP
+export KMP_BLOCKTIME=1
+export KMP_SETTINGS=1
+export KMP_AFFINITY=granularity=fine,compact,1,0
+
+# OMP
+export OMP_NUM_THREADS=56
```
Comment (hshen14) [inline intel_extension_for_transformers/neural_chat/examples/retrieval/run_retrieval_on_cpu.sh:24]: better to use <core count> # physical cores on single socket

File: intel_extension_for_transformers/neural_chat/examples/retrieval/run_retrieval_on_cpu.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+export LD_PRELOAD=${CONDA_PREFIX}/lib/libiomp5.so
+
+# tc malloc
+export LD_PRELOAD=${LD_PRELOAD}:${CONDA_PREFIX}/lib/libtcmalloc.so
+
+numactl -l -C 0-55 python retrieval_chat.py
```
Comment (hshen14) [inline intel_extension_for_transformers/neural_chat/examples/retrieval/run_retrieval_on_cpu.sh:30]: 0-55 => <cpu list>

**Comment by hshen14** [comment]
> the name of "run_retrieval_on_cpu.sh" may lead to unexpected confusion when the device has both GPU and CPU. Need to rename or specify the device in the code.

I am assuming CPU here is the target device. If this is the case, the script name should be ok.

**Comment by hshen14** [comment]
@lvliang-intel  please fix the format issues

**Review by XuhuiRen — APPROVED** [review]

File: intel_extension_for_transformers/neural_chat/requirements.txt [hunk 3]
```diff
@@ -33,3 +33,4 @@ sentence_transformers
 unstructured
 markdown
 rouge_score
+openpyxl
```
Comment (hshen14) [inline intel_extension_for_transformers/neural_chat/requirements.txt:36]: please confirm with Wenxin whether it has been added BOM

Comment (XuhuiRen) [reply intel_extension_for_transformers/neural_chat/requirements.txt:36]: > please confirm with Wenxin whether it has been added BOM

I left a message to Wenxin about this package. It's license is MIT license. Should be ok

**Pull request merged**